src/ca/team2706/vision/trackerboxreloaded/Main.java

```diff
@@ -306,34 +411,49 @@ public static void main(String[] args) {
         // Main video processing loop
         while (true) {
             if (useCamera) {
+            	//Read the frame from the camera, if it fails try again
                 if (!camera.read(frame)) {
                     System.err.println("Error: Failed to get a frame from the camera");
                     continue;
                 }
             } // else use the image from disk that we loaded above
-
-            if (use_GUI)
-                Imgproc.resize( frame, frame, visionParams.sz );
+            if(use_GUI)
+            //Resize the frame
+            Imgproc.resize( frame, frame, visionParams.sz );
 
             // Process the frame!
+            //Log when the pipeline starts
             long pipelineStart = System.nanoTime();
+            //Process the frame
             VisionData visionData = Pipeline.process(frame, visionParams, use_GUI);
+            //Log when the pipeline stops
             long pipelineEnd = System.nanoTime();
-
+            //Selects the prefered target
             Pipeline.selectPreferredTarget(visionData, visionParams);
-
-            Mat outputTargetImg = frame.clone();
-            Pipeline.drawPreferredTarget(outputTargetImg, visionData);
-
+            //Creates the raw output image object
+            Mat rawOutputImg;
+            if (use_GUI) {
+            	//If use gui then draw the prefered target
+                //Sets the raw image to the frame
+            	rawOutputImg = frame.clone();
+                //Draws the preffered target
+                Pipeline.drawPreferredTarget(rawOutputImg, visionData);
+            } else {
+                //Sets the raw image to the frame
+            	rawOutputImg = frame;
+            }
+            //Sends the data to the vision table
             sendVisionDataOverNetworkTables(visionData);
 
             // display the processed frame in the GUI
             if (use_GUI) {
                 try {
                     // May throw a NullPointerException if initializing
                     // the window failed
-                    guiRawImg.updateImage(matToBufferedImage(outputTargetImg));
-                    guiProcessedImg.updateImage(matToBufferedImage(visionData.binMask));
+                	//Update the raw image gui
+                    guiRawImg.updateImage(matToBufferedImage(rawOutputImg));
+                    //Update the processed image gui
+                    guiProcessedImg.updateImage(matToBufferedImage(visionData.outputImg));
```

Comment:
This is also a code change (`outputTargetImg` --> `rawOutputImg`, and `binMask` --> `outputImg`). From memory, I think the `binMask` version is actually correct. This was a change that I made after Ryerson, which I'm guessing wasn't on whichever branch you started your `Documentation` branch from.